src/cpu.rs

```diff
@@ -63,8 +71,109 @@ impl CPU {
+                        // (This is sign extended to a 16-but data type, but an unsigned one: u16. It's a
+                        // little weird, but it's so we can add the PC and the offset easily)
+                        let offset = slice[0];
+                        let sign_extend = if offset & 0x80 == 0x80 { 0xffu8 } else { 0x0 };
+                        let rel = u16::from_le_bytes([offset, sign_extend]);
+                        OpInput::UseRelative(rel)
+                    }
+                    AddressingMode::Absolute => {
+                        // Use [u8, ..2] from instruction as address
+                        // (Output: a 16-bit address)
+                        OpInput::UseAddress(arr_to_addr(&slice))
+                    }
+                    AddressingMode::AbsoluteX => {
+                        // Use [u8, ..2] from instruction as address, add X
+                        // (Output: a 16-bit address)
+                        OpInput::UseAddress(arr_to_addr(&slice).wrapping_add(xextend(x)))
+                    }
+                    AddressingMode::AbsoluteY => {
+                        // Use [u8, ..2] from instruction as address, add Y
+                        // (Output: a 16-bit address)
+                        OpInput::UseAddress(arr_to_addr(&slice).wrapping_add(xextend(y)))
+                    }
+                    AddressingMode::Indirect => {
+                        // Use [u8, ..2] from instruction as an address. Interpret the
+                        // two bytes starting at that address as an address.
+                        // (Output: a 16-bit address)
+                        let slice = read_address(memory, arr_to_addr(&slice));
+                        OpInput::UseAddress(arr_to_addr(&slice))
+                    }
+                    AddressingMode::IndexedIndirectX => {
+                        // Use [u8, ..1] from instruction
+                        // Add to X register with 0-page wraparound, like ZeroPageX.
+                        // This is where the absolute (16-bit) target address is stored.
+                        // (Output: a 16-bit address)
+                        let start = slice[0].wrapping_add(x);
+                        let slice = read_address(memory, u16::from(start));
+                        OpInput::UseAddress(arr_to_addr(&slice))
+                    }
+                    AddressingMode::IndirectIndexedY => {
+                        // Use [u8, ..1] from instruction
+                        // This is where the absolute (16-bit) target address is stored.
+                        // Add Y register to this address to get the final address
+                        // (Output: a 16-bit address)
+                        let start = slice[0];
+                        let slice = read_address(memory, u16::from(start));
+                        OpInput::UseAddress(arr_to_addr(&slice).wrapping_add(xextend(y)))
+                    }
+                };
```

Comment:
Good code. If I had to nitpick, I might suggest extracting the match statement into a separate method or module, to reduce the amount of code in this function. But that's more of a matter of personal preference

Reply:
Yeah originally this was all in a separate method but I ran into [this problem](https://github.com/mre/mos6502/issues/50#issuecomment-1254569333) so the body of that method has basically been moved here.